x-pack/plugins/canvas/public/components/workpad_app/workpad_app.component.tsx

```diff
@@ -38,12 +48,20 @@ export const WorkpadApp: FC<Props> = ({ deselectElement, isWriteable }) => {
 
   const commit = interactivePageLayout.current || (() => {});
 
+  const untitledWorkpadLabel = getUntitledWorkpadLabel();
+
   return (
     <div className="canvasLayout">
       <div className="canvasLayout__rows">
         <div className="canvasLayout__cols">
           <div className="canvasLayout__stage">
             <div className="canvasLayout__stageHeader">
+              <h1
+                id="canvasWorkpadTitle"
+                className="euiScreenReaderOnly"
+                ref={workpadTitle}
+                tabIndex={-1}
+              >{`${CANVAS} - ${workpad.name || untitledWorkpadLabel}`}</h1>
```

Comment:
**nit:** Could we just do `${CANVAS} - ${workpad.name || getUntitledWorkpadLabel()}` so that we only fetch the untitled workpad label if it's necessary rather than fetching it on line `51`?

Reply:
I thought it was bad practice to invoke a function in the JSX because it causes unnecessary re-renders, but what I can do is wrap the variable in a `useMemo` to reduce the re-renders.

Reply:
Ahh, very fair point 👍 Didn't know that, good info! :)